src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeployedToEnvironmentNotifierStep.java

```diff
@@ -0,0 +1,242 @@
+package com.atlassian.bitbucket.jenkins.internal.deployments;
+
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeploymentEnvironment;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeploymentEnvironmentType;
+import com.atlassian.bitbucket.jenkins.internal.provider.JenkinsProvider;
+import com.atlassian.bitbucket.jenkins.internal.scm.BitbucketSCMRepository;
+import com.atlassian.bitbucket.jenkins.internal.status.BitbucketRevisionAction;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Item;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Notifier;
+import hudson.tasks.Publisher;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+import javax.annotation.CheckForNull;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.stripToNull;
+
+public class DeployedToEnvironmentNotifierStep extends Notifier implements SimpleBuildStep {
+
+    private static final FormValidation FORM_VALIDATION_OK = FormValidation.ok();
+    private static final Logger LOGGER = Logger.getLogger(DeployedToEnvironmentNotifierStep.class.getName());
+
+    private final BitbucketDeploymentEnvironment environment;
+
+    @DataBoundConstructor
+    public DeployedToEnvironmentNotifierStep(String environmentKey, String environmentName,
+                                             @CheckForNull String environmentType,
+                                             @CheckForNull String environmentUrl) {
+        environment = validateAndGetEnvironment(environmentKey, environmentName, environmentType, environmentUrl);
+    }
+
+    public DescriptorImpl descriptor() {
+        return Jenkins.get().getDescriptorByType(DescriptorImpl.class);
+    }
+
+    public BitbucketDeploymentEnvironment getEnvironment() {
```

Comment:
are the following methods only used for tests? Or are they also going to be used by other classes in the future?
I feel like we can just keep `getEnvironment` since the other methods are accessors of the `environment`

Reply:
The other methods are actually used by Jenkins to populate the fields on the page when you edit the job